catalog/resourcemanagement/resourcemanagement-usage-ui/src/main/java/org/codice/ddf/resourcemanagement/usage/service/DataUsage.java

```diff
@@ -0,0 +1,260 @@
+                String id = (String) preference.get(ID + TXT_PREFIX);
+                if (CRON_TIME_ID_KEY.equals(id)) {
+                    return (String) preference.get(CRON_TIME_KEY + TXT_PREFIX);
+                }
+            }
+        } catch (PersistenceException e) {
+            LOGGER.error("Unable to get Cron Time from Persistent Store.", e);
+        }
+        return DEFAULT_CRON_TIME;
+    }
+
+    private void setPersistentCronTime() {
+        PersistentItem persistentItem = new PersistentItem();
+        persistentItem.addIdProperty(CRON_TIME_ID_KEY);
+        persistentItem.addProperty(CRON_TIME_KEY, this.cronTime);
+
+        try {
+            readWriteLock.readLock()
+                    .lock();
+            persistentStore.add(PREFERENCES_TYPE, persistentItem);
+
+        } catch (PersistenceException e) {
+            LOGGER.error("Error adding Cron Time to Persistent Store.", e);
+        } finally {
+            readWriteLock.readLock()
+                    .unlock();
+        }
+    }
+
+    private void registerMbean() {
+        try {
+            objectName = new ObjectName(DataUsage.class.getName() + ":service=datausage");
+            mBeanServer = ManagementFactory.getPlatformMBeanServer();
+        } catch (MalformedObjectNameException e) {
+            LOGGER.error("Unable to create Data Usage Configuration MBean.", e);
+        }
+        if (mBeanServer == null) {
+            return;
+        }
+        try {
+            try {
+                mBeanServer.registerMBean(this, objectName);
+                LOGGER.info("Registered Data Usage Configuration MBean under object name: {}",
+                        objectName.toString());
+            } catch (InstanceAlreadyExistsException e) {
+                // Try to remove and re-register
+                mBeanServer.unregisterMBean(objectName);
+                mBeanServer.registerMBean(this, objectName);
+                LOGGER.info("Re-registered Data Usage Configuration MBean");
+            }
+        } catch (MBeanRegistrationException | InstanceNotFoundException |
+                InstanceAlreadyExistsException | NotCompliantMBeanException e) {
+            LOGGER.error("Could not register MBean [{}].", objectName.toString(), e);
+        }
+
+    }
+
+    @Override
+    public Map<String, List<Long>> userMap() {
+        Map<String, List<Long>> dataUsageMap = new HashMap<>();
+        try {
+
+            attributesStore.getAllUsers()
+                    .stream()
+                    .filter(MapUtils::isNotEmpty)
+                    .forEach(stringObjectMap -> {
+                                if (MapUtils.isNotEmpty(stringObjectMap)) {
+                                    dataUsageMap.put((String) stringObjectMap.get(
+                                                    AttributesStore.USER_KEY + TXT_PREFIX),
+                                            Arrays.asList((Long) stringObjectMap.get(
```

Comment:
Within the context of this code, it is guaranteed that these objects will be `Long`